### ISSUE TITLE
Abseil build with -Werror on Bazel fails if using clang-9 or older

### ISSUE THREAD

**Issue by ofats** [abseil/abseil-cpp#1007]
**Describe the bug**

Abseil fails to build if using `-Werror` on older **clang** because of unknown warning options.
For example, `-Wno-implicit-float-conversion` is unknown on **clang-9**.

**Steps to reproduce the bug**

```bash
# On the repo root
$ bazel build ... --action_env=CC=clang-9 --copt=-Werror
INFO: Build option --copt has changed, discarding analysis cache.
INFO: Analyzed 388 targets (0 packages loaded, 2078 targets configured).
INFO: Found 388 targets...
ERROR: <workspace>/absl/hash/BUILD.bazel:119:11: Compiling absl/hash/internal/city.cc failed: (Exit 1): clang-9 failed: error executing command /usr/bin/clang-9 -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer '-std=c++0x' -MD -MF ... (remaining 51 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
error: unknown warning option '-Wno-implicit-int-float-conversion'; did you mean '-Wno-implicit-float-conversion'? [-Werror,-Wunknown-warning-option]
INFO: Elapsed time: 10.103s, Critical Path: 5.84s
INFO: 21 processes: 9 internal, 12 linux-sandbox.
FAILED: Build did NOT complete successfully
```

**What version of Abseil are you using?**

https://github.com/abseil/abseil-cpp/commit/637722af3a60c17915d3325604a0435ee92a41b4 (master when creating this issue)

**What operating system and version are you using**

```bash
$ uname -a
5.4.0-74-generic #83-Ubuntu SMP Sat May 8 02:35:39 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

**What compiler and version are you using?**

```bash
$ clang-9 -v
clang version 9.0.1-12 
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/9
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/9
Selected GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/9
Candidate multilib: .;@m64
Selected multilib: .;@m64
```

**What build system are you using?**

```bash
$ bazel --version
bazel 4.2.0
```

**Additional context**

I see that you're using `-Werror` option when building with clang in your CI:
https://github.com/abseil/abseil-cpp/blob/637722af3a60c17915d3325604a0435ee92a41b4/ci/linux_clang-latest_libcxx_bazel.sh#L86
It would have been discovered on older clang easily if you don't use only the latest version.

**Comment by ofats** [comment]
Fixing in #1008

**Comment by derekmauro** [comment]
The fix in #1008 has been merged.
